enumerables.rb

```diff
@@ -1,88 +1,100 @@
+# frozen_string_literal: true
+
+# enumerable methods
 module Enumerable
   def my_each
     arr = self
+    if block_given?
       for i in arr
         yield(i)
+        puts i
       end
+    else
+      arr.to_enum(:my_each)
+    end
   end
 
   def my_each_with_index
     arr = self
-      for i in arr
+    if block_give?
+      arr.my_each do |i|
         v = arr.find_index(i)
         yield(i, v)
       end
+    else
+      arr.to_enum(:my_each_with_index)
+    end
   end
 
   def my_select
     arr = self
     result = []
-    for i in arr
-      if  yield i
-        result.push(i)
-        end
+    arr.each do |i|
+      result.push(i) if yield i
     end
     result
   end
 
-  def my_all(argument = nil)
+  def my_all?(arg = nil)
     arr = self
     if block_given?
       arr.my_each do |i|
-      return false unless yield(i)
+        return false unless yield(i)
       end
     else
-      my_each { |i| return false if i.nil? || i == false }
+      my_each { |i| return false if !i.is_a? arg }
     end
     true
   end
-
-  def my_all(argument = nil)
+  def my_any?( arg = nil )
     arr = self
     if block_given?
-      arr.my_each do |i|
-      return false
+      arr.my_each do |i| 
+        return true if yield(i)
       end
     else
-      my_each { |i| yield i if i.nil? || i == false }
+      my_each { |i| return true if i.is_a? arg }
     end
-    true
+    false
   end
 
-  def my_none(argument = nil)
+  def my_none?( arg = nil )
     arr = self
     if block_given?
       arr.my_each do |i|
-      return false if yield i
+        return false if yield i
       end
     else
-      my_each { |i| yield i if i.nil? && i == false }
+      my_each { |i| return false if i.is_a? arg }
     end
     true
   end
```

Comment:
* [ ] should have working `my_none?` identical and returns the same thing as ruby's [`none?`](https://ruby-doc.org/core-2.6.4/Enumerable.html#method-i-none-3F). ([Screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/ruby/advanced_building_blocks_enumerable/my_none.png) from [Odin](https://www.theodinproject.com/courses/ruby-programming/lessons/advanced-building-blocks#assignment-2))

* [ ] my_none? when no block or argument is given returns true only if none of the collection members is true
  ```ruby
    array = [nil, false, nil, false]
    p array.my_none? == array.none? # false
  ```
* [ ] my_none? when a Regex is passed as an argument returns true only if none of the collection matches the Regex
  ```ruby
    array = %w[dog door rod blade]
    p array.my_none?(/c/) == array.none?(/c/) # false
  ```

* [ ] my_none? when a pattern other than Regex or a Class is given returns true only if none of the collection matches the pattern
  ```ruby
    array = %w[dog door rod blade]
    p array.my_none?(5) == array.none?(5) # false
  ```